packages/DateTimeInput/src/index.tsx

```diff
@@ -0,0 +1,111 @@
+import withUtils from "@blaze-react/utils";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import DatePicker from "react-datepicker";
+import { DATE_FORMAT_MAP, TYPE_DATE, TYPE_DATE_TIME, TYPE_TIME } from './constants';
+
+import "react-datepicker/dist/react-datepicker.css";
+
+interface IErrorMessage {
+  message: string | JSX.Element;
+  icon?: string;
+}
+
+interface IDateTimeInputProps {
+  disabled?: boolean;
+  id?: string;
+  label?: string;
+  modifier?: string;
+  onChange: ({
+    event,
+    value,
+  }: {
+    event: React.ChangeEvent<HTMLInputElement>;
+    value: Date;
+  }) => void;
+  required?: boolean;
+  error?: boolean;
+  type?: string;
+  validationMessage: string | JSX.Element;
+  value?: Date;
+  utils: {
+    classNames: (className: string | object, classNames?: object) => string;
+    ErrorMessage: FunctionComponent<IErrorMessage>;
+  };
+}
+
+const DateTimeInput: FunctionComponent<IDateTimeInputProps> = ({
+  disabled,
+  id,
+  label,
+  modifier,
+  onChange,
+  required,
+  error,
+  type,
+  validationMessage,
+  value,
+  utils: { classNames, ErrorMessage },
+}): JSX.Element => {
+  const [newValue, setNewValue] = useState<Date | undefined>(value);
+  const [newError, setError] = useState<boolean | undefined>(error);
+
+  let whitelistedType: string = type as string;
+  if (type !== TYPE_TIME && type !== TYPE_DATE) {
+    whitelistedType = TYPE_DATE_TIME;
+  }
+
+  useEffect(() => {
+    setError(error);
+  }, [error]);
+
+  useEffect(() => {
+    setNewValue(value)
+  }, [value]);
+
+  const handleChange = (date: Date, event: React.ChangeEvent<HTMLInputElement>): void => {
+    setNewValue(date);
+    onChange({ event, value: date });
+  };
+
+  const requiredClassName: string = classNames({ required });
+
+  const modifierClassName: string = classNames({
```

Comment:
Another way to do it, to avoid using template string
```
const modifierClassName: string = classNames('form-field form-field--date-time-input', {
    [`form-field--${modifier}`]: !!modifier,
  });

<div
      className={modifierClassName}
    >
```